azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineImages.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.azure.management.compute;
+
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
+import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
+import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
+import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+
+public interface VirtualMachineImages extends
+        SupportsListing<VirtualMachine>,
+        SupportsListingByGroup<VirtualMachine>,
+        SupportsGettingByGroup<VirtualMachine>,
+        SupportsCreating<VirtualMachine>,
+        SupportsDeleting,
+        SupportsDeletingByGroup {
+    interface InGroup extends
+            SupportsListing<VirtualMachine>,
+            SupportsCreating<VirtualMachine>,
+            SupportsDeleting {}
+}
```

Comment:
The template parameter type should be "VirtualMachineImage" instead of "VirtualMachine"